### PR TITLE
feat: add option --smart-group

### PR DESCRIPTION
**Pull request by liketurbo**
Relevant issue: #480

### CODE REVIEW

**Review by cafkafk — CHANGES_REQUESTED** [review]
You'll need to add this to zsh, fish, and bash completions, as well as the man page.

**Comment by liketurbo** [comment]
> You'll need to add this to zsh, fish, and bash completions, as well as the man page.

I probably didn't do bash completions correctly, as there isn't much to use as a template ... so I'd gladly accept any help with that one 😊

File: completions/bash/eza [hunk 1]
```diff
@@ -4,7 +4,7 @@ _eza() {
     prev=${COMP_WORDS[COMP_CWORD-1]}
 
     case "$prev" in
-        --help|-v|--version)
+        --help|-v|--version|--smart-group)
```
Comment (cafkafk) [inline completions/bash/eza:7]: (Just reminding myself to check this before merging)

**Review by cafkafk — CHANGES_REQUESTED** [review]

**Review by cafkafk — APPROVED** [review]
Manpage generation works, integration tests work, and the expected behavior is present without breaking other functionality (as far as I can tell).

**Pull request merged**